src/hooks/useUserVote.ts

```diff
@@ -0,0 +1,27 @@
+import { TESTNET_CONTRACTS } from "@/lib/contractConstants";
+import { useNear } from "@/contexts/NearContext";
+import { useReadHOSContract } from "./useReadHOSContract";
+
+export function useUserVote(proposalId: number) {
+  const { signedAccountId } = useNear();
+
+  const [result] = useReadHOSContract([
+    {
+      contractId: TESTNET_CONTRACTS.VOTING_CONTRACT_ID,
+      methodName: "get_vote",
+      config: {
+        args: {
+          account_id: signedAccountId || "",
+          proposal_id: proposalId,
+        },
+        enabled: !!signedAccountId,
+      },
+    },
+  ]);
+
+  return {
+    voteIndex: result.data as number | null | undefined,
```

Comment:
We shouldn't need to type cast this - the `useReadHOSContract()` hook should type the data correctly for us. Is that not the case here? If so, we might need to cast the `methodName` to be `as const` to get the typing to work